From: Sergio Costas Rodriguez <sergio.costas@canonical.com>
Date: Wed, 18 Jun 2025 17:25:29 +0200
Subject: a11y: Add role to the panel list elements

The elements in the panel list lack an a11y role. This patch
sets their role to `button` to notify that they are clickables.

Origin: https://gitlab.gnome.org/GNOME/gnome-control-center/-/merge_requests/3187
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnome-control-center/+bug/2115973
---
 shell/cc-panel-list.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/shell/cc-panel-list.c b/shell/cc-panel-list.c
index f715131..8c46889 100644
--- a/shell/cc-panel-list.c
+++ b/shell/cc-panel-list.c
@@ -289,6 +289,10 @@ row_data_new (CcPanelCategory     category,
                                   label,
                                   NULL,
                                   -1);
+  g_object_set (G_OBJECT (data->row),
+                "accessible-role",
+                GTK_ACCESSIBLE_ROLE_BUTTON,
+                NULL);
 
   gtk_widget_add_css_class (label, "dim-label");
   gtk_grid_attach (GTK_GRID (grid), label, 1, 1, 1, 1);
